plugins/processors/scale/scale.go

```diff
@@ -0,0 +1,119 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package Scale
+
+import (
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Scale) SampleConfig() string {
+	return sampleConfig
+}
+
+type Scaling struct {
+	InMin  float64  `toml:"input_minimum"`
+	InMax  float64  `toml:"input_maximum"`
+	OutMin float64  `toml:"output_minimum"`
+	OutMax float64  `toml:"output_maximum"`
+	Fields []string `toml:"fields"`
+
+	factor      float64
+	fieldFilter filter.Filter
+}
+
+type Scale struct {
+	Scalings []Scaling       `toml:"scaling"`
+	Log      telegraf.Logger `toml:"-"`
+}
+
+func (s *Scaling) init() error {
+	if s.InMax == s.InMin {
+		return fmt.Errorf("input minimum and maximum are equal for fields %s", strings.Join(s.Fields, ","))
+	}
+
+	if s.OutMax == s.OutMin {
+		return fmt.Errorf("output minimum and maximum are equal for fields %s", strings.Join(s.Fields, ","))
+	}
+
+	scalingFilter, err := filter.Compile(s.Fields)
+	if err != nil {
+		return fmt.Errorf("could not compile filter: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("could not compile fields filter: %w", err)
```

Maybe also indicate to which fields filter it applies?